reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportsSchedulerRestHandler.java

```diff
@@ -0,0 +1,161 @@
+ * {@code POST /_opendistro/scheduler_sample/watch?id=kibana-job-id&job_name=watch kibana index&index=.kibana_1&interval=1"}
+ *
+ * creates a job with id "1" and job name "watch kibana index", which logs ".kibana_1" index's shards info
+ * every 1 minute
+ *
+ * Users can remove that job by calling
+ * {@code DELETE /_opendistro/scheduler_sample/watch?id=kibana-job-id}
+ */
+public class ReportsSchedulerRestHandler extends BaseRestHandler {
+    public static final String SCHEDULER_URI = "/_opendistro/reports_scheduler/schedule";
+
+    @Override
+    public String getName() {
+        return "Reports Scheduler Handler";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return Collections.unmodifiableList(Arrays.asList(
+                new Route(RestRequest.Method.POST, SCHEDULER_URI),
+                new Route(RestRequest.Method.DELETE, SCHEDULER_URI)
+        ));
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        if (request.method().equals(RestRequest.Method.POST)) {
+            // compose JobParameter object from request
+            String id = request.param("id");
+
+            //TODO: add start_time and Time_unit from request params or request body
+
+            //TODO: parse the content not params.
+            /**
+             * {
+             *
+             *     "name": "Flights monthly report",
+             *     "enabled": true, //boolean, default=true, required by ScheduledJobParameter
+             *     "schedule": {
+             *         "interval": {
+             *             "period": 1,
+             *             "unit": "MINUTES",
+             *             "start_time": 1553112384
+             *         }
+             *
+             *       },
+             *     "report_definition_id":"23dasdfdasdas"
+             *  }
+             */
+
+            if(id == null) {
+                throw new IllegalArgumentException("Must specify id");
+            }
+
+            IndexRequest indexRequest = new IndexRequest()
+                    .index(ReportsSchedulerPlugin.JOB_INDEX_NAME)
+                    .id(id)
+                    .source(request.requiredContent(), XContentType.JSON);
+
+            return restChannel -> {
+                // index the job parameter
+                client.index(indexRequest, new ActionListener<IndexResponse>() {
+                    @Override
+                    public void onResponse(IndexResponse indexResponse) {
+                        try {
+                            RestResponse restResponse = new BytesRestResponse(RestStatus.OK,
+                                    indexResponse.toXContent(JsonXContent.contentBuilder(), null));
+                            restChannel.sendResponse(restResponse);
+                        } catch(IOException e) {
+                            restChannel.sendResponse(new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, e.getMessage()));
```

Comment:
is it always internal server error if not success!?

Reply:
Yes, the action is save data to elasticsearch index, by using the client